presto-docs/src/main/sphinx/develop/web-ui.rst

```diff
@@ -0,0 +1,81 @@
+======
+Web UI
+======
+
+Presto comes with a built-in dashboard where you can check the cluster’s information, running SQL queries,
+query details, resource group information, etc. Presto web UI comprises a series of HTML pages and JavaScript
+files which invokes the Presto APIs served by the coordinator. The UI components are written in
+`React JS <https://react.dev/>`_ and built by `webpack <https://webpack.js.org/>`_. Here are the guidances
+for web UI development.
+
+Code
+----
+
+The web UI code can be found in the ``presto-main/src/main/resources/webapp`` directory in the root of the
+Presto source tree. This directory is referred to as ``web UI root`` in this document and contains:
+
+- HTML files: the static pages for the web UI.
+- ``assets`` folder: contains Presto favicon, logo, and Presto CSS file.
+- ``dev`` directory: contains ``Query Viewer`` page and its JS file.
+- ``dist`` directory: contains the JS files generated by webpack.
+- ``vendor`` directory: contains third-party JS/CSS libraries that are used by UI components.
+- ``src`` directory: contains the UI components and webpack config file.
+
+Prerequisites
+-------------
+
+- `yarn <https://yarnpkg.com/>`_: A package manager for the Node.js project. It's used to manage the packages
+  used by the UI components and the scripts to compile and generate the JavaScript files.
+
+UI components
+-------------
+
+All of the UI components are in the ``src`` folder under the ``web UI root`` directory.  These components are
+written in `JSX <https://react.dev/learn/writing-markup-with-jsx>`_ using React JS and named with the ``.jsx``
+file extension.
+
+Please follow these rules for composing UI components:
+
+- Define components as JavaScript functions instead of classes: Some old components are written in classes and
+  will be refactored into functions. For new components, please use function components. You can check
+  `here <https://react.dev/learn/your-first-component#defining-a-component>`_ for detailed information.
```

Comment:
```suggestion
  `Defining a component <https://react.dev/learn/your-first-component#defining-a-component>`_ for detailed information.
```
See Gitlab doc style guide [Text for links](https://docs.gitlab.com/ee/development/documentation/styleguide/#text-for-links) for context.